submariner/cable-driver-protocols-enhancement.md

```diff
@@ -0,0 +1,76 @@
+# Cable Driver Enhancement - IP-in-IP
+
+## Summary
+
+Submariner today supports three cable drivers: VXLAN, IPSec (tunnel mode) and WireGuard. The
+goal of this proposal is to outline additional enhancements needed for the Submariner Cable Driver
+to support of IP-in-IP for unencrypted connections between clusters or for use as an overlay with
+IPSec in transport mode for encrypted connections between clusters.
+
+For IP in IP, the benefits fall into the following categories:
+
+* **Performance**: IP in IP is has a lower overhead than VXLAN which could lead to performance
+improvements in terms of packet overhead (bandwidth consumption) as well as the cost of processing
+packets. Some of the performance improvements are shown below [1]:
+
+  * ![Throughput rate vs Kernel](./images/tx-tput-rate.png)
+  * ![Throughput rate Gbps](./images/tx-tput-gbps.png)
+  * ![CPU cycles](./images/cpu-cycles.png)
+
+* **Interoperability**: IP in IP can be used to transport packets between domains when the protocol
+in those domains (e.g. IPv6) is not supported by intermediary networks (e.g. IPv4).
+
+## Proposal
+
+This enhancement proposes to support IP-in-IP for un-encrypted connection between clusters. A
+new IP tunnelling interface `ipip-tunnel` will be added to the active gateway node of each cluster
+and it will act as the Tunnel End Point. The traffic which is destined to other clusters will be
+forwarded via this interface to the appropriate clusters.
+
+> **_NOTE:_** A single IP Tunnelling interface will be added, and tunnels managed using
+`# ip route add encap ip <args>` in order to avoid a point to point connection per tunnel.
+
+### Design Details
+
+* A new interface "ipip-tunnel" will be added.
```

Comment:
Might be a good idea to include some reference to "submariner" (like sm-ipip-tunnel) in the interface name to convey that its managed by Submariner. I know that we are using "vxlan-tunnel" in case of VxLAN cable driver, but it was more of an oversight and we missed it.